packages/server/graphql/types/GitLabSearchQuery.ts

```diff
@@ -14,11 +15,10 @@ const GitLabSearchQuery = new GraphQLObjectType<any, GQLContext>({
       type: new GraphQLNonNull(GraphQLString),
       description: 'The query string used to search GitLab issue titles and descriptions'
     },
-    selectedProjectsIds: {
-      type: new GraphQLList(new GraphQLNonNull(GraphQLID)),
+    selectedProjects: {
```

Comment:
In the "Current Filters", we want to display a list of the selected GitLab project `fullPath`s: 

![image](https://user-images.githubusercontent.com/39854876/161792763-88606f92-8421-4784-8fd0-e9fc49ba9fa3.png)

There are a few ways that we could get the projects `fullPath`s:

1. Current approach: in the `GitLabSearchQuery`, we can change `selectedProjectIds` to `selectedProjects`, which contains the `id`s and `fullPath`s for each project. This is easy to query on the frontend, but in GitLab v2, when we store persisted queries in the db, we'll need to store an extra field, `fullPath`, in the db. 
2. In [GitLabIntegration](https://github.com/ParabolInc/parabol/blob/f39ccbb0bc4fa3b680668c74985ff6e36f060140/packages/server/graphql/types/GitLabIntegration.ts), we could store a `projects` field. In the resolver, we could query projects, in a similar way to how we query an issue in Task ([see here](https://github.com/ParabolInc/parabol/blob/4712bec016a670388622caae3f8653f3b1f74f69/packages/server/graphql/types/Task.ts#L217)). The downside is we would be sending a query to GitLab to get all projects here, in the projects filter menu, and in the create new issue dropdown menu. 
3. Like the [AtlassianIntegration](https://github.com/ParabolInc/parabol/blob/072cdfcbe5cf1a5eed3a9c72ec1da51756d2b9fe/packages/server/graphql/types/AtlassianIntegration.ts#L163), we could create a GitLab custom dataLoader that grabs all projects. The custom dataLoader would be used here, the projects filter menu, and in the create new issue dropdown menu. As the dataLoader caches the results, we wouldn't need to reach out to GitLab each time, which would make the integration feel more snappy.

I'm thinking the third approach is best, but I want to get a second opinion before implementing it. Perhaps there's a fourth option that's even better 🤷‍♂️

Reply:
Storing `fullPath` isn't great because that value is mutable, ie GitLab users can rename their repo. We should try to only store immutable keys that we can exchange for fresh data from the integration, even if that means an extra round trip, it's better than stale data! (I used to store our github repo name in the DB & then we changed parabolinc/action to parabolinc/parabol & it was stale forever)

Option 3 wouldn't work because a dataloader only caches per-request & the filter & create new issue dropdown are separate requests. You could cache the projects in redis if you were concerned about latency, but then you'd have to verify that the value in redis has all the fields that you require (e.g. if you cached `{id, title}` in redis and then a 2nd request tries to get `{id, fullPath}`, you'd need a way to tell it to go back out to gitlab.

Option 2 is actually more efficient than you might think. You can use relay's `store-or-network` mode to only trigger a fetch if there is missing data. So as long as your 2nd query is a subset of your first query, there won't be a 2nd network request.

Reply:
That all makes sense, thanks! I was thinking that as `fullPath` is unique, it essentially acts as an id, but you're right, users can just rename it 😄 

I've pushed an update that implements option 2 👍